src/panels/lovelace/cards/energy/hui-energy-compare-card.ts

```diff
@@ -46,6 +54,10 @@ export class HuiEnergyCompareCard
     ];
   }
 
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    return hasConfigChanged(this, changedProps) || changedProps.size > 1;
```

Comment:
I think this works better?
```suggestion
    return hasConfigChanged(this, changedProps) || changedProps.size > 1 || !changedProps.has("hass");
```

Reply:
`hasConfigChanged` already returns true if changedProps.get("hass") is undefined, so I think this is not necessary?

```
  const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
  if (!oldHass) {
    return true;
  }
```

I thought that choice was a bit strange, maybe that was unintentional?

Reply:
That seems like a bug, it probably wants to check if there is no `oldHass`, so it is the first time `hass` is set, then it would need to update ofc, but it is missing the `changedProps.hass("hass")` check...

Reply:
https://github.com/home-assistant/frontend/pull/18778

Reply:
Added explicit !has("hass") checks.